install.sh

```diff
@@ -0,0 +1,198 @@
+    if [ -z "$(command -v brew)" ]; then
+        NONINTERACTIVE=1 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+        if [[ uname -m == "x86_64" ]]; then
+            echo "export PATH=/usr/local/bin:$PATH" >> ~/.bash_profile && source ~/.bash_profile
+        else
+            echo "export PATH=/opt/homebrew/bin:$PATH" >> ~/.bash_profile && source ~/.bash_profile
+        fi
+    else
+        echo "Homebrew is already installed"
+    fi
+    echo "Installing python"
+    if ! command -v python3 &> /dev/null; then
+        brew install python@3.10
+    else
+        echo "Python is already installed"
+    fi    
+}
+
+function install_arch(){
+    echo "Installing python"
+    if [ ! command -v python3 &> /dev/null ]; then
+        sudo pacman -S python-pip
+    else
+        echo "Python is already installed"
+    fi
+    if [ ! command -v git &> /dev/null ]; then
+        sudo pacman -S git
+    else
+        echo "Git is already installed"
+    fi
+}
+
+function install_deb(){
+    echo "Installing python"
+    if [ ! command -v python3 &> /dev/null ]; then
+        sudo apt-get install python3
+    else
+        echo "Python is already installed"
+    fi
+    if [ ! command -v pip3 &> /dev/null ]; then
+        sudo apt-get install python3-pip
+    else
+        echo "pip is already installed"
+    fi
+    if [ ! command -v git &> /dev/null ]; then
+        sudo apt-get install git
+    else
+        echo "git is already installed"
+    fi
+}
+
+function install_fedora(){
+    echo "Installing python"
+    if [ ! command -v python3 &> /dev/null ]; then
+        sudo dnf install python3
+    else
+        echo "Python is already installed"
+    fi
+    if [ ! command -v pip3 &> /dev/null ]; then
+        sudo dnf install python3-pip
+    else
+        echo "pip is already installed"
+    fi
+    if [ ! command -v git &> /dev/null ]; then
+        sudo dnf install git
+    else
+        echo "git is already installed"
+    fi
+}
+
+function install_centos(){
+    echo "Installing python"
+    if [ ! command -v python3 &> /dev/null ]; then
+        sudo yum install python3
+    else
+        echo "Python is already installed"
+    fi
+    if [ ! command -v pip3 &> /dev/null ]; then
+        python3 -m ensurepip --upgrade
+    else
+        echo "pip is already installed"
+    fi
+    if [ ! command -v git &> /dev/null ]; then
+        sudo yum install git
+    else
+        echo "git is already installed"
+    fi
+}
+
+function get_the_bot(){
+    echo "Downloading the bot"
+    git clone https://github.com/elebumm/RedditVideoMakerBot.git
+}
+
+function install_python_dep(){
+    echo "Installing python dependencies"
+    cd RedditVideoMakerBot
+    pip3 install -r requirements.txt
+}
+
+function install_playwright(){
+    echo "Installing playwright"
+    cd RedditVideoMakerBot
+    python3 -m playwright install
+    python3 -m playwright install-deps
+}
+
+install_deps(){
```

Comment:
This isn't a great way of using the correct package manager - what if someone is using a fork of a distro? They would still have the package manager. Something like
```sh
if [ -x "$(command -v apk)" ];       then sudo apk add --no-cache $packages
elif [ -x "$(command -v apt-get)" ]; then sudo apt-get install $packages
elif [ -x "$(command -v dnf)" ];     then sudo dnf install $packages
elif [ -x "$(command -v zypper)" ];  then sudo zypper install $packages
elif [ -x "$(command -v yum)" ];  then sudo yum install $packages
else echo "FAILED TO INSTALL PACKAGE: Package manager not found. You must manually install: $packagesNeeded">&2; fi
```
would work better.

Reply:
Greta, i never acctualky tougth of that. Will fix once i get back to my pc